src/system/NetworkNamespace.cpp

```diff
@@ -21,36 +21,49 @@ std::mutex namespacesLock;
 
 void returnNetworkNamespace(std::shared_ptr<NetworkNamespace> ns)
 {
+    if (ns->getMode() == NetworkIsolationMode::ns_off) {
+        return;
+    }
     faabric::util::UniqueLock lock(namespacesLock);
     namespaces.emplace_back(ns);
 }
 
 std::shared_ptr<NetworkNamespace> claimNetworkNamespace()
 {
+    faabric::util::UniqueLock lock(namespacesLock);
     if (namespaces.empty() && !namespacesInitialised) {
-        faabric::util::UniqueLock lock(namespacesLock);
+        conf::FaasmConfig& conf = conf::getFaasmConfig();
+        int nNamespaces = conf.maxNetNs;
+
+        // If network namespaces are disabled, overwrite the env. variable and
+        // create a single pointer to use as a placeholder
+        if (conf.netNsMode == "off") {
```

Comment:
I can't immediately tell the difference between these two lines:

```
if (ns->getMode() == NetworkIsolationMode::ns_off) {
 ```
 
 and 
 
 ```
if (conf.netNsMode == "off") {
 ```
 
 Is there any? Seems like only one form  of this check should be used.